### DISCUSSION TITLE
Custom WebDriverAgent installation

### DISCUSSION THREAD

**Discussion by jobeylev** [danielpaulus/go-ios#211] [Q&A]
Hi, I need to sign and deploy WebDriverAgent as a standalone IPA, so that our engineers can install WDA onto iPhones connected to their windows machines. I managed to create what I think is a valid IPA, and my test engineers are able to install it from Windows to their iPhones. However, when they try to run WDA, they get this error:

 WebDriverAgentRunner-Runner[78215:3278635] Failed to retrieve test configuration from IDE. Exiting. Error: (null)

This error does not occur if I run the same IPA on an iPhone that already has our in-house signed WDA installed. I suspect this means Xcode does something magical to it, adding some configuration that's otherwise missing?

Is this related to the --xctestconfig setting on the runwda command? Perhaps I don't understand how this is used. I use it to point to the WebDriverAgentRunner.xctest plugin. If that's not right, can you tell me what kind of file it should be pointing to, and how I can generate a file like that?

**Comment by admFIINEX** [comment]
Hi same problem here, did you pass him ?

**Comment by nanoscopic** [comment]
Any WDA like thing must be recursively signed properly in order for xctest launching of it to function. If only the outer wrapper was signed you can install it but the xctest portion of the app will not function.

It's probable that you have not used proper tooling able to correctly sign/re-sign the WDA you are attempting to use.

There is nothing "magical" about this, it just isn't well understood publicly and Apple certainly doesn't explain this as they don't provide any tooling to resign existing app+xctest. ( or resign at all for that matter )

And yes when using go-ios I think you may need to create and use a 'xctestconfig' as well; but someone else will need to comment on that as I don't actually use go-ios to launch xctests myself any more...
